sample.py

```diff
@@ -0,0 +1,58 @@
+# This function adds two numbers
+def add(x, y):
+    return x + y
+
+
+# This function subtracts two numbers
+def subtract(x, y):
+    return x - y
+
+
+# This function multiplies two numbers
+def multiply(x, y):
+    return x * y
+
+
+# This function divides two numbers
+def divide(x, y):
+    return x / y
+
+
+print("Select operation.")
+print("1.Add")
+print("2.Subtract")
+print("3.Multiply")
+print("4.Divide")
+
+while True:
+    # take input from the user
+    choice = input("Enter choice(1/2/3/4): ")
+
+    # check if choice is one of the four options
+    if choice in ('1', '2', '3', '4'):
+        try:
+            num1 = float(input("Enter first number: "))
+            num2 = float(input("Enter second number: "))
+        except ValueError:
+            print("Invalid input. Please enter a number.")
+            continue
```

Comment:
<div>


<div id="suggestion">
<div id="issue"><b>Consider more granular error handling</b></div>
<div id="fix">

The error handling for invalid number inputs could be more specific by catching `ValueError` separately for each input.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -32,10 +32,16 @@
     if choice in ('1', '2', '3', '4'):
         try:
             num1 = float(input("Enter first number: "))
 -            num2 = float(input("Enter second number: "))
          except ValueError:
 -            print("Invalid input. Please enter a number.")
 +            print("Invalid input for first number. Please enter a number.")
 +            continue
 +        
 +        try:
 +            num2 = float(input("Enter second number: "))
 +        except ValueError:
 +            print("Invalid input for second number. Please enter a number.")
              continue

```
</div>
</details>
</div>


<small><i>Code Review Run <a href=https://github.com/Amruta101998/DS-and-Algorithm/pull/29#issuecomment-2514890238>#56bb45</a></i></small>
</div>

---
Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged